src/main/java/uk/gov/hmcts/cmc/claimstore/documents/BulkPrintHandler.java

```diff
@@ -84,4 +88,28 @@ public void print(GeneralLetterReadyToPrintEvent event) {
             GENERAL_LETTER_TYPE
         );
     }
+
+    @EventListener
+    public void print(BulkPrintTransferEvent event) {
+        requireNonNull(event);
+        Claim claim = event.getClaim();
+
+        PrintablePdf coverLetter = new PrintablePdf(
+            event.getCoverLetter(),
+            buildCoverSheetFileBaseName(claim.getReferenceNumber()));
+
+        ImmutableList.Builder<Printable> printableDocs = ImmutableList.<Printable>builder().add(coverLetter);
+        addCaseDocuments(event.getCaseDocuments(), printableDocs);
+
+        bulkPrintService.printPdf(claim, printableDocs.build(), BULK_PRINT_TRANSFER_TYPE);
+    }
+
+    private void addCaseDocuments(
+        List<BulkPrintTransferEvent.PrintableDocument> caseDocuments,
+        ImmutableList.Builder<Printable> printableDocs
+    ) {
+        for (BulkPrintTransferEvent.PrintableDocument document : caseDocuments) {
+            printableDocs.add(new PrintablePdf(document.getDocument(), document.getFileName()));
+        }
+    }
```

Comment:
This might be a bit better than passing the builder around.
```suggestion
       List<Printable> printableDocs = List.of(coverLetter);
       printableDocs.addAll(event.getCaseDocuments()
           .stream()
           .map(d -> new PrintablePdf(d.getDocument(), d.getFileName()))
           .collect(Collectors.toList())
       );
        
        bulkPrintService.printPdf(claim, Collections.unmodifiableList(printableDocs), BULK_PRINT_TRANSFER_TYPE);
    }   
```